app/screens/apps_form/apps_form_component.tsx

```diff
@@ -149,7 +153,10 @@ export default class AppsFormComponent extends PureComponent<Props, State> {
             submission.values[this.props.form.submit_buttons] = button;
         }
 
+        this.setState({submitting: true});
```

Comment:
I know previous reviwers approved, but why is it needed to setState and re-render the component? I don't see the button being disabled. Isn't enough just using a class variable?

Reply:
I foresee we reintroducing this in the future when we use buttons instead of the navigation bar, but for the time being it makes more sense to do that, yes.